lib/dispatcher.js

```diff
@@ -14,6 +18,20 @@ class Dispatcher extends EventEmitter {
   destroy () {
     throw new Error('not implemented')
   }
+
+  intercept (...interceptors) {
+    let dispatcher = this
+    for (const interceptor of interceptors) {
+      if (typeof interceptor !== 'function') {
+        throw new Error('invalid interceptor')
+      }
+      dispatcher = interceptor(dispatcher) ?? dispatcher
+      if (dispatcher[kDispatcherVersion] !== 1) {
+        throw new Error('invalid dispatcher')
+      }
```

Comment:
It's mostly if we e.g. change the intercept method in an incompatible way. I'm not using instanceof on purpose so that it works if there are different version of undici at the same time.